miio/viomivacuum.py

```diff
@@ -268,124 +382,400 @@ def has_map(self) -> bool:
+
+    # @property
+    # def order_time(self) -> int:
+    #    """FIXME: ??? int or bool."""
+    #    return self.data["order_time"]
+
+    @property
+    def repeat_cleaning(self) -> bool:
+        """Secondary clean up state.
+
+        True if the cleaning is performed twice
+        """
+        return self.data["repeat_state"]
+
+    # @property
+    # def start_time(self) -> int:
+    #    """FIXME: ??? int or bool."""
+    #    return self.data["start_time"]
+
+    @property
+    def voice_state(self) -> ViomiVoiceState:
+        """Voice volume level (from 0 to 100%, 0 means Off)."""
+        return ViomiVoiceState(self.data["v_state"])
+
+    # @property
+    # def water_percent(self) -> int:
+    #    """FIXME: ??? int or bool."""
+    #    return self.data["water_percent"]
+
+    # @property
+    # def zone_data(self) -> int:
+    #    """FIXME: ??? int or bool."""
+    #    return self.data["zone_data"]
+
+
+def _get_rooms_from_schedules(schedules: List[str]) -> Tuple[bool, Dict]:
+    """Read the result of "get_ordertime" command to extract room names and ids.
+
+    The `schedules` input needs to follow the following format
+    * ['1_0_32_0_0_0_1_1_11_0_1594139992_2_11_room1_13_room2', ...]
+    * [Id_Enabled_Repeatdays_Hour_Minute_?_? _?_?_?_?_NbOfRooms_RoomId_RoomName_RoomId_RoomName_..., ...]
+
+    The function parse get_ordertime output to find room names and ids
+    To use this function you need:
+    1. to create a scheduled cleanup with the following properties:
+    * Hour: 00
+    * Minute: 00
+    * Select all (minus one) the rooms one by one
+    * Set as inactive scheduled cleanup
+    2. then to create an other scheduled cleanup with the room missed at
+       previous step with the following properties:
+    * Hour: 00
+    * Minute: 00
+    * Select only the missed room
+    * Set as inactive scheduled cleanup
+
+    More information:
+    * https://github.com/homebridge-xiaomi-roborock-vacuum/homebridge-xiaomi-roborock-vacuum/blob/d73925c0106984a995d290e91a5ba4fcfe0b6444/index.js#L969
+    * https://github.com/homebridge-xiaomi-roborock-vacuum/homebridge-xiaomi-roborock-vacuum#semi-automatic
+
+
+    >>> _parse_schedules(['1_0_32_0_0_0_1_1_11_0_1594139992_2_11_room1_13_room2'])
+    (True, {'11': 'room1', '13': 'room2'})
+    >>> _parse_schedules(['1_0_32_0_0_0_1_1_11_0_1594139992_2_11_room1_13_room2', '1_0_32_0_0_0_1_1_11_0_1594139992_2_10_room3_14_room4'])
+    (True, {'11': 'room1', '13': 'room2', '10': 'room3', '14': 'room4'})
+    >>> _parse_schedules(['1_1_32_0_0_0_1_1_11_0_1594139992_2_11_room1_13_room2', '1_0_32_0_0_0_1_1_11_0_1594139992_2_10_room3_14_room4'])
+    (True, {'10': 'room3', '14': 'room4'})
+    >>> _parse_schedules(['1_0_32_2_0_0_1_1_11_0_1594139992_2_11_room1_13_room2', '1_0_32_0_4_0_1_1_11_0_1594139992_2_10_room3_14_room4'])
+    (False, {})
+    >>> _parse_schedules(['1_1_32_0_0_0_1_1_11_0_1594139992_2_11_room1_13_room2'])
+    (False, {})
+    >>> _parse_schedules([])
+    (False, {})
```

Comment:
Instead of having such in docstrings, you could add some unittests to the parser for these cases. That would keep the code-base cleaner, and making sure no future changes will break the behavior when/if the support will be improved in the future!

Reply:
done